api/src/main/java/jakarta/mvc/form/FormMethodOverwriter.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2022 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License v. 2.0, which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception, which is available at
+ * https://www.gnu.org/software/classpath/license.html.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR GPL-2.0 WITH Classpath-exception-2.0
+ */
+package jakarta.mvc.form;
+
+/**
+ * <p>Form method overwriter are used to overwrite a HTML form's HTTP method to be able to use
+ * other verbs such as {@code PATCH} or {@code DELETE}. They are discovered using Jakarta Contexts and Dependency Injection.
+ * </p>
+ * 
+ * @author Tobias Erdle
+ * @since 2.1
+ */
+public final class FormMethodOverwriter {
```

Comment:
> Is it really just about HTML forms?

My first impulse is to say "Yes", as JavaScript clients could target the API by using the correct method directly. But that depends, as always, on the client developer. So personally I'm not attached to the _form_ inside the class name.

> Maybe we could use overwrite instead of overwriter?

Yes, absolutely. This was a leftover of the initial draft.

Reply:
> My first impulse is to say "Yes", as JavaScript clients could target the API by using the correct method directly. But that depends, as always, on the client developer. So personally I'm not attached to the form inside the class name.

After thinking a bit more about this: Actually, even in case of PPR requests, the request would submit a form, although from JavaScript using `FormData` or some other mechanism. On the other hand: In theory, an MVC Controller method could also process some kind of JSON document sent via POST and return a view in HTML, or am I missing something?